tests/testsuite/bad_config.rs

```diff
@@ -2151,6 +2151,37 @@ Caused by:
         .run();
 }
 
+#[cargo_test]
+fn github_pull_request_url() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.0.0"
+                edition = "2015"
+                authors = []
+
+                [dependencies.bar]
+                git = "https://github.com/foo/bar/pull/123"
+            "#,
+        )
+        .file("src/lib.rs", "")
+        .build();
+
+    p.cargo("check -v")
+        .with_status(101)
+        .with_stderr_data(str![[r#"
+[ERROR] failed to parse manifest at `[ROOT]/foo/Cargo.toml`
+
+Caused by:
+  dependency (bar) specifies a GitHub pull request link. If you were trying to specify a specific github PR, replace the URL with the git URL (e.g. `git = "https://github.com/foo/bar.git"`) and add `rev = "refs/pull/123/head"` in the dependency declaration.
```

Comment:
As an error message, it should first saw what the problem is.  I also feel like this requires people to parse a lot of the message to understand the fix and apply it (e.g. rather than copy/pasting)

What do you think of
```
  `dependencies.{name}.git = "{url}"` is not a repository.
  hint: The repository path looks like a pull request, try replacing the repository path with:
    `git = "{url}", rev = "{rev}"`
```

Reply:
The first part is for consistency with the other error messages. How about:
```
dependency (bar) git url https://github.com/foo/bar/pull/123 is not a repository.
The path looks like a pull request.
Try replacing the dependency with: `git = "https://github.com/foo/bar.git" rev = "refs/pull/123/head"` in the dependency declaration.
```

Reply:
I am fine with keeping it consistent with existing messages. We could tweak them later altogether when needed.